### PR TITLE
Include gl3d in strict bundle

### PR DESCRIPTION
**Pull request by archmoj**
Addresses #897 and adds all `gl3d` traces as well as `heatmapgl` and `pointcloud` traces from `gl2d` subplots to the "strict" bundle.

To test it within dashboard after checking out the branch
```sh
npm ci
npm run strict
```

@plotly/plotly_js

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -105,22 +107,29 @@
     "gl-pointcloud2d": "^1.0.3",
     "gl-scatter3d": "^1.2.3",
     "gl-select-box": "^1.0.4",
+    "gl-shader": "^4.3.1",
```
Comment (archmoj) [inline package.json:110]: https://github.com/stackgl/gl-shader/pull/22

File: package.json [hunk 2]
```diff
@@ -130,15 +139,22 @@
     "regl-scatter2d": "^3.2.6",
     "regl-splom": "^1.0.14",
     "right-now": "^1.0.0",
-    "robust-orientation": "^1.1.3",
+    "robust-determinant": "git://github.com/archmoj/robust-determinant.git#0ea5b0b45ae73e01e2789934b6bfbf029c243563",
+    "robust-in-sphere": "^1.2.1",
+    "robust-linear-solve": "git://github.com/archmoj/robust-linear-solve.git#0cd1ce1ff68da9b96786e383aba0ff78777be119",
+    "robust-orientation": "^1.2.1",
```
Comment (archmoj) [inline package.json:145]: https://github.com/mikolalysenko/robust-orientation/pull/5

File: package.json [hunk 3]
```diff
@@ -130,15 +139,22 @@
...
+    "simplicial-complex-contour": "git://github.com/archmoj/simplicial-complex-contour.git#5ca6fe1e36642e25a451d6b39534939f7f229e5d",
     "strongly-connected-components": "^1.0.1",
     "superscript-text": "^1.0.0",
+    "surface-nets": "git://github.com/archmoj/surface-nets.git#805acfc4c52a1c9a6aaf989cb98ffd5136001896",
     "svg-path-sdf": "^1.1.3",
     "tinycolor2": "^1.4.2",
     "to-px": "1.0.1",
     "topojson-client": "^3.1.0",
+    "vectorize-text": "^3.2.2",
```
Comment (archmoj) [inline package.json:154]: https://github.com/mikolalysenko/vectorize-text/pull/19

File: package.json [hunk 4]
```diff
@@ -130,15 +139,22 @@
     "regl-scatter2d": "^3.2.6",
     "regl-splom": "^1.0.14",
     "right-now": "^1.0.0",
-    "robust-orientation": "^1.1.3",
+    "robust-determinant": "git://github.com/archmoj/robust-determinant.git#0ea5b0b45ae73e01e2789934b6bfbf029c243563",
+    "robust-in-sphere": "^1.2.1",
```
Comment (archmoj) [inline package.json:143]: https://github.com/mikolalysenko/robust-in-sphere/pull/2

File: package.json [hunk 5]
```diff
@@ -130,15 +139,22 @@
...
+    "robust-linear-solve": "git://github.com/archmoj/robust-linear-solve.git#0cd1ce1ff68da9b96786e383aba0ff78777be119",
```
Comment (archmoj) [inline package.json:144]: https://github.com/mikolalysenko/robust-linear-solve/pull/1

File: package.json [hunk 6]
```diff
@@ -130,15 +139,22 @@
     "regl-scatter2d": "^3.2.6",
     "regl-splom": "^1.0.14",
     "right-now": "^1.0.0",
-    "robust-orientation": "^1.1.3",
+    "robust-determinant": "git://github.com/archmoj/robust-determinant.git#0ea5b0b45ae73e01e2789934b6bfbf029c243563",
```
Comment (archmoj) [inline package.json:142]: https://github.com/mikolalysenko/robust-determinant/pull/1

File: package.json [hunk 7]
```diff
@@ -105,22 +107,29 @@
...
     "gl-select-box": "^1.0.4",
+    "gl-shader": "^4.3.1",
     "gl-spikes2d": "^1.0.2",
     "gl-streamtube3d": "^1.4.1",
     "gl-surface3d": "^1.6.0",
     "gl-text": "^1.2.0",
     "glslify": "^7.1.1",
     "has-hover": "^1.0.1",
     "has-passive-events": "^1.0.0",
+    "incremental-convex-hull": "git://github.com/archmoj/incremental-convex-hull.git#f34c97097239a25f69cca0f7252b1a8517876f9f",
```
Comment (archmoj) [inline package.json:118]: https://github.com/mikolalysenko/incremental-convex-hull/pull/1

File: package.json [hunk 8]
```diff
@@ -130,15 +139,22 @@
     "regl-scatter2d": "^3.2.6",
     "regl-splom": "^1.0.14",
     "right-now": "^1.0.0",
-    "robust-orientation": "^1.1.3",
+    "robust-determinant": "git://github.com/archmoj/robust-determinant.git#0ea5b0b45ae73e01e2789934b6bfbf029c243563",
+    "robust-in-sphere": "^1.2.1",
+    "robust-linear-solve": "git://github.com/archmoj/robust-linear-solve.git#0cd1ce1ff68da9b96786e383aba0ff78777be119",
+    "robust-orientation": "^1.2.1",
+    "simplicial-complex-contour": "git://github.com/archmoj/simplicial-complex-contour.git#5ca6fe1e36642e25a451d6b39534939f7f229e5d",
```
Comment (archmoj) [inline package.json:146]: https://github.com/mikolalysenko/simplicial-complex-contour/pull/2

File: package.json [hunk 9]
```diff
@@ -130,15 +139,22 @@
...
     "strongly-connected-components": "^1.0.1",
     "superscript-text": "^1.0.0",
+    "surface-nets": "git://github.com/archmoj/surface-nets.git#805acfc4c52a1c9a6aaf989cb98ffd5136001896",
```
Comment (archmoj) [inline package.json:149]: https://github.com/mikolalysenko/surface-nets/pull/2

File: package.json [hunk 10]
```diff
@@ -77,6 +78,7 @@
     "@turf/bbox": "^6.4.0",
     "@turf/centroid": "^6.0.2",
     "alpha-shape": "^1.0.0",
+    "box-intersect": "git://github.com/archmoj/box-intersect.git#89622c6e9cadad8f442b3faa6dad2bc56e400d40",
```
Comment (archmoj) [inline package.json:81]: https://github.com/mikolalysenko/box-intersect/pull/8

File: package.json [hunk 11]
```diff
@@ -105,22 +107,29 @@
...
     "mouse-change": "^1.4.0",
     "mouse-event-offset": "^3.0.2",
     "mouse-wheel": "^1.2.0",
     "native-promise-only": "^0.8.1",
-    "ndarray": "^1.0.19",
+    "ndarray": "git://github.com/archmoj/ndarray.git#210ccfbf32d085f40603a8b44f81d50ddf2b7d83",
+    "ndarray-extract-contour": "git://github.com/archmoj/ndarray-extract-contour.git#563317603ac0e4ad203f197cfdd989f60da94e39",
+    "ndarray-gradient": "git://github.com/archmoj/ndarray-gradient.git#28f575c2719eed56c48e29df8bf318e1111ef082",
     "ndarray-linear-interpolate": "^1.0.0",
+    "ndarray-ops": "git://github.com/archmoj/ndarray-ops.git#ce9969aa054262e6df6d54c2a07a39fcf7b7bb6d",
```
Comment (archmoj) [inline package.json:130]: https://github.com/scijs/ndarray-ops/pull/13

File: package.json [hunk 12]
```diff
@@ -69,6 +69,7 @@
     ]
   },
   "dependencies": {
+    "3d-view": "git://github.com/archmoj/3d-view.git#96b0d7d2176b4c60256bb05404dfddfb02e7485d",
```
Comment (archmoj) [inline package.json:72]: https://github.com/mikolalysenko/3d-view/pull/1

File: package.json [hunk 13]
```diff
@@ -105,22 +107,29 @@
...
     "mapbox-gl": "1.10.1",
     "matrix-camera-controller": "^2.1.4",
     "mouse-change": "^1.4.0",
     "mouse-event-offset": "^3.0.2",
     "mouse-wheel": "^1.2.0",
     "native-promise-only": "^0.8.1",
-    "ndarray": "^1.0.19",
+    "ndarray": "git://github.com/archmoj/ndarray.git#210ccfbf32d085f40603a8b44f81d50ddf2b7d83",
+    "ndarray-extract-contour": "git://github.com/archmoj/ndarray-extract-contour.git#563317603ac0e4ad203f197cfdd989f60da94e39",
+    "ndarray-gradient": "git://github.com/archmoj/ndarray-gradient.git#28f575c2719eed56c48e29df8bf318e1111ef082",
```
Comment (archmoj) [inline package.json:128]: https://github.com/scijs/ndarray-gradient/pull/1

File: package.json [hunk 14]
```diff
@@ -105,22 +107,29 @@
...
+    "incremental-convex-hull": "git://github.com/archmoj/incremental-convex-hull.git#f34c97097239a25f69cca0f7252b1a8517876f9f",
     "is-mobile": "^2.2.2",
     "mapbox-gl": "1.10.1",
     "matrix-camera-controller": "^2.1.4",
     "mouse-change": "^1.4.0",
     "mouse-event-offset": "^3.0.2",
     "mouse-wheel": "^1.2.0",
     "native-promise-only": "^0.8.1",
-    "ndarray": "^1.0.19",
+    "ndarray": "git://github.com/plotly/ndarray.git#210ccfbf32d085f40603a8b44f81d50ddf2b7d83",
```
Comment (archmoj) [inline package.json:126]: https://github.com/plotly/ndarray/pull/1

File: package.json [hunk 15]
```diff
@@ -105,22 +107,29 @@
...
+    "ndarray-extract-contour": "git://github.com/plotly/ndarray-extract-contour.git#563317603ac0e4ad203f197cfdd989f60da94e39",
+    "ndarray-gradient": "git://github.com/plotly/ndarray-gradient.git#28f575c2719eed56c48e29df8bf318e1111ef082",
     "ndarray-linear-interpolate": "^1.0.0",
+    "ndarray-ops": "git://github.com/plotly/ndarray-ops.git#ce9969aa054262e6df6d54c2a07a39fcf7b7bb6d",
```
Comment (archmoj) [inline package.json:130]: https://github.com/plotly/ndarray-ops/pull/1

File: package.json [hunk 16]
```diff
@@ -105,22 +107,29 @@
...
+    "ndarray-pack": "git://github.com/plotly/ndarray-pack.git#4a51cd5c0ba02a3717893021aba1df69830595b6",
+    "ndarray-sort": "git://github.com/plotly/ndarray-sort.git#0cc8b110e397e94cdb208ec26d04f29f5b3bb2eb",
```
Comment (archmoj) [inline package.json:132]: https://github.com/plotly/ndarray-sort/pull/1

File: package.json [hunk 17]
```diff
@@ -105,22 +107,29 @@
...
     "mapbox-gl": "1.10.1",
     "matrix-camera-controller": "^2.1.4",
     "mouse-change": "^1.4.0",
     "mouse-event-offset": "^3.0.2",
     "mouse-wheel": "^1.2.0",
     "native-promise-only": "^0.8.1",
-    "ndarray": "^1.0.19",
+    "ndarray": "git://github.com/plotly/ndarray.git#210ccfbf32d085f40603a8b44f81d50ddf2b7d83",
+    "ndarray-extract-contour": "git://github.com/plotly/ndarray-extract-contour.git#563317603ac0e4ad203f197cfdd989f60da94e39",
+    "ndarray-gradient": "git://github.com/plotly/ndarray-gradient.git#28f575c2719eed56c48e29df8bf318e1111ef082",
```
Comment (archmoj) [inline package.json:128]: https://github.com/plotly/ndarray-gradient/pull/1

File: package.json [hunk 18]
```diff
@@ -130,15 +139,22 @@
...
+    "surface-nets": "git://github.com/archmoj/surface-nets.git#805acfc4c52a1c9a6aaf989cb98ffd5136001896",
     "svg-path-sdf": "^1.1.3",
     "tinycolor2": "^1.4.2",
     "to-px": "1.0.1",
     "topojson-client": "^3.1.0",
+    "vectorize-text": "^3.2.2",
     "webgl-context": "^2.2.0",
-    "world-calendars": "^1.0.3"
+    "world-calendars": "^1.0.3",
+    "zero-crossings": "git://github.com/plotly/zero-crossings.git#bb8ea62d1014c3d8a1aa6e5d6332abf4dc371f94"
```
Comment (archmoj) [inline package.json:157]: https://github.com/plotly/zero-crossings/pull/1

File: package.json (hunk 10 above)
Comment (archmoj) [reply package.json:81]: https://github.com/plotly/box-intersect/pull/1

File: package.json (hunk 9 above)
Comment (archmoj) [reply package.json:149]: https://github.com/plotly/surface-nets/pull/1

File: package.json (hunk 8 above)
Comment (archmoj) [reply package.json:146]: https://github.com/plotly/simplicial-complex-contour/pull/1

File: package.json (hunk 7 above)
Comment (archmoj) [reply package.json:118]: https://github.com/plotly/incremental-convex-hull/pull/1

File: package.json (hunk 6 above)
Comment (archmoj) [reply package.json:142]: https://github.com/plotly/robust-determinant/pull/1

File: package.json (hunk 5 above)
Comment (archmoj) [reply package.json:144]: https://github.com/plotly/robust-linear-solve/pull/1

File: package.json (hunk 12 above)
Comment (archmoj) [reply package.json:72]: https://github.com/plotly/3d-view/pull/1

File: package.json [hunk 19]
```diff
@@ -152,7 +168,7 @@
     "deep-equal": "^2.0.5",
     "derequire": "^2.1.1",
     "ecstatic": "^4.1.4",
-    "eslint": "^7.32.0",
+    "eslint": "git://github.com/archmoj/eslint#083ad40001b45ad06b07ef0449b6f06e5c9bb099",
```
Comment (archmoj) [inline package.json:171]: https://github.com/eslint/eslint/pull/14860

File: package.json [hunk 20]
```diff
@@ -105,22 +107,29 @@
...
+    "incremental-convex-hull": "plotly/incremental-convex-hull.git#v1.1.0",
     "is-mobile": "^2.2.2",
     "mapbox-gl": "1.10.1",
     "matrix-camera-controller": "^2.1.4",
     "mouse-change": "^1.4.0",
     "mouse-event-offset": "^3.0.2",
     "mouse-wheel": "^1.2.0",
     "native-promise-only": "^0.8.1",
-    "ndarray": "^1.0.19",
+    "ndarray": "plotly/ndarray.git#v1.1.0",
```
Comment (alexcjohnson) [inline package.json:126]: Looking at https://docs.npmjs.com/cli/v7/configuring-npm/package-json#github-urls it appears the `.git` can (should?) also be omitted.

Comment (archmoj) [reply package.json]: Done in https://github.com/plotly/plotly.js/pull/5888/commits/84b3a23cdcc0f4c45fcc48c5a1d24586ae6ded63.

**Review by alexcjohnson — APPROVED** [review]
Excellent! I'm not going to go through all the new code you've added in all the sub-packages... the tests pass so I'm happy :) 💃

**Comment by archmoj** [comment]
Changes to the bundle sizes: 
- Red: before
- Green: after

sizes

This could be reduced in follow up PR(s) by removing unused dimensions e.g. [iso-surface generation](https://github.com/plotly/surface-nets/blob/3f66a813cb54b31ea8de47ab242549e22370a51e/surfacenets.js#L97-L1556) in `surface-nets` and other modules.

**Pull request merged**